pulp_python/app/serializers.py

```diff
@@ -247,27 +242,22 @@ def deferred_validate(self, data):
                 "Extension on {} is not a valid python extension "
                 "(.whl, .exe, .egg, .tar.gz, .tar.bz2, .zip)").format(filename)
             )
+        if data.get("sha256") and data["sha256"] != artifact.sha256:
+            raise serializers.ValidationError(
+                detail={"sha256": _("This field must match uploaded artifact's sha256")}
```

Comment:
I think this error message could be improved.  Something along the lines of "The uploaded artifact's sha256 checksum does not match the one provided"